crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4312,6 +4330,18 @@ pub struct BoundMethodType<'db> {
     self_instance: Type<'db>,
 }
 
+impl<'db> BoundMethodType<'db> {
+    pub(crate) fn into_callable_type(self, db: &'db dyn Db) -> Type<'db> {
+        let signature = self.function(db).signature(db);
+        let parameters = Parameters::new(signature.parameters().iter().skip(1).cloned());
+
+        Type::Callable(CallableType::General(GeneralCallableType::new(
+            db,
+            Signature::new(parameters, signature.return_ty),
+        )))
```

Comment:
Can you move the part that manipulates the `Signature` into a new `Signature::bind_self` method? That will make it a bit more readable here:

```suggestion
        Type::Callable(CallableType::General(GeneralCallableType::new(
            db,
            self.function(db).signature(db).bind_self(),
        )))
```

Reply:
Sure